swarm/src/lib.rs

```diff
@@ -1112,6 +1113,37 @@ where
                     self.pool.disconnect(peer_id);
                 }
             },
+            ToSwarm::NewExternalAddrOfPeer(addr) => {
+                log::debug!("ToSwarm::NewExternalAddrOfPeer");
+
+                let translated_addresses = {
+                    let mut addrs: Vec<_> = self
+                        .listened_addrs
+                        .values()
+                        .flatten()
+                        .filter_map(|server| self.transport.address_translation(server, &addr))
+                        .collect();
```

Comment:
Given that this is the address of a remote and not a local address, I don't think there is any value in `address_translation`. Likely just a copy and paste from the `NewAddrCandidate` event.

Reply:
:heavy_check_mark: 

Left just `on_swarm_event` call

Reply:
Sounds good. Let me know once you need another review here.